dev/Interop/CameraCaptureUI/CameraCaptureUI/CameraCaptureUI.cpp

```diff
@@ -0,0 +1,137 @@
+static constexpr const wchar_t* CameraPickerUri = L"microsoft.windows.camera.picker:";
+
+
+namespace winrt::Microsoft::Windows::Media::Capture::implementation
+{
+    IAsyncOperation<hstring> LaunchCameraForResultToken(winrt::Microsoft::UI::WindowId const& targetWindow, ValueSet const& properties)
+    {
+        LauncherOptions options;
+        if (targetWindow.Value)
+        {
+            check_hresult(options.as<IInitializeWithWindow>()->Initialize(reinterpret_cast<HWND>(targetWindow.Value)));
+        }
+
+        options.TreatAsUntrusted(false);
+        options.DisplayApplicationPicker(false);
+        options.TargetApplicationPackageFamilyName(CameraAppPackageName);
+
+        Uri launchUri(CameraPickerUri);
+        winrt::Windows::System::LaunchUriResult result = co_await Launcher::LaunchUriForResultsAsync(launchUri, options, properties);
+        if (!result.Result())
+        {
+            co_return L"";
+        }
+        else if (auto t = result.Result().TryLookup(L"SelectedTokens"))
+        {
+            co_return t.as<hstring>();
+        }
+        else
+        {
+            throw hresult_canceled();
+        }
+    }
+
+    concurrency::task<token_and_path> CreateEmptyFileAndGetToken(hstring const& fileExtension)
+    {
+        auto tempFolder = ApplicationData::Current().TemporaryFolder();
+        auto tempFile = co_await tempFolder.CreateFileAsync(L"CameraCapture" + fileExtension, CreationCollisionOption::GenerateUniqueName);
+        auto tempFilePath = tempFile.Path();
+        auto token = SharedStorageAccessManager::AddFile(tempFile);
+
+        co_return token_and_path{ token, tempFilePath };
+    }
+
+    IAsyncOperation<StorageFile> CameraCaptureUI::CaptureFileAsync(CameraCaptureUIMode mode)
+    {
+        // Only allow one capture operation at a time from this instance
+        std::unique_lock<std::mutex> lock(m_captureMutex, std::try_to_lock);
+        if (!lock.owns_lock())
+        {
+            throw hresult_error(E_ILLEGAL_METHOD_CALL, L"CameraCaptureUI is already capturing");
+        }
+
+        bool isAppPackaged = m_telemetryHelper.IsPackagedApp();
+        PCWSTR appName = m_telemetryHelper.GetAppName().c_str();
+        const wchar_t* mediaType = nullptr;
+        try
+        {
+            auto strong = get_strong();
+
+            if (mode == CameraCaptureUIMode::PhotoOrVideo)
+            {
+                mediaType = L"photoOrVideo";
+            }
+            else if (mode == CameraCaptureUIMode::Photo)
+            {
+                mediaType = L"photo";
+            }
+            else if (mode == CameraCaptureUIMode::Video)
+            {
+                mediaType = L"video";
+            }
+            else
+            {
+                throw hresult_invalid_argument();
+            }
+            auto logCaptureOperation{ CameraCaptureUITelemetry::CameraCaptureOperation::Start(isAppPackaged, appName, mediaType) };
```

Comment:
You can move the start event up.

Reply:
We explicitly create a string for the mode, so it makes more sense to use it. Also, if someone passes a value other than photo, video, or photo mode, the operation won't start. I think it functions correctly as is!

Reply:
You can pass mode as enum directly as received from caller in the activity, no need to wait for its conversion to string. Early you declare the activity better it is to capture all kinds of errors i.e. it should cover incorrect mode too.